src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs

```diff
@@ -771,26 +842,28 @@ public ICommand DeleteProfileCommand
                     {
                         var profileToRemove = SelectedDebugProfile;
                         SelectedDebugProfile = null;
-                        DebugProfiles.Remove(profileToRemove);
-                        SelectedDebugProfile = DebugProfiles.Count > 0 ? DebugProfiles[0] : null;
+                        LaunchProfiles.Remove(profileToRemove);
+                        SelectedDebugProfile = LaunchProfiles.Count > 0 ? LaunchProfiles[0] : null;
+                        NotifyProfileCollectionChanged();
                     }));
             }
         }
 
-        internal LaunchProfile CreateProfile(string name, string commandName)
+        internal void CreateProfile(string name, string commandName)
         {
-            var profile = new LaunchProfile() { Name = name, CommandName = commandName };
-            DebugProfiles.Add(profile);
+            var profile = new WritableLaunchProfile() { Name = name, CommandName = commandName };
+            CurrentLaunchSettings.Profiles.Add(profile);
 
-            // Fire a property changed so we can get the page to be dirty when we add a new profile
+            NotifyProfileCollectionChanged();
+            
+            // Todo: Do we need this extra property change
```

Comment:
I put back the original comment. Not sure why I added the todo in the first place.